lib/metasploit/framework/login_scanner/teamcity.rb

```diff
@@ -0,0 +1,181 @@
+          }
+
+          opts.each { |param, value| request_params[param] = value }
+          begin
+            res = send_request(request_params)
+          rescue ::Rex::ConnectionError, ::Rex::ConnectionProxyError, ::Errno::ECONNRESET, ::Errno::EINTR, ::Rex::TimeoutError, ::Timeout::Error, ::EOFError => e
+            return { status: UNABLE_TO_CONNECT, proof: e }
+          end
+
+          return { status: UNABLE_TO_CONNECT, proof: 'Unable to connect to the TeamCity service' } if res.nil?
+          # Does the service need to be setup & configured with the initial DB migration & admin account?
+          return { status: UNABLE_TO_CONNECT, proof: "Received an unexpected status code: #{res.code}. Does the service need to be configured?" } if res.code != 200
+
+          { status: :success, proof: res }
+        end
+
+        # Extract the server's public key from the response.
+        # @param [Rex::Proto::Http::Response] response The response to extract the public RSA key from.
+        # @return [Hash] A hash with a status and an error or the server's public key.
+        def get_public_key(response)
+          html_doc = response.get_html_document
+          public_key_choices = html_doc.xpath('//input[@id="publicKey"]/@value')
+          return { status: UNABLE_TO_CONNECT, proof: 'Could not find the TeamCity public key in the HTML document' } if public_key_choices.empty?
+
+          { status: :success, proof: public_key_choices.first.value }
+        end
+
+        # Create a login request body for the provided credentials.
+        # @param [String] username The username to create the request body for.
+        # @param [String] password The user's password.
+        # @param [Object] public_key The public key to use when encrypting the password.
+        def create_login_request_body(username, password, public_key)
+          vars = {}
+          vars['username'] = URI.encode_www_form_component(username)
+          vars['remember'] = 'true'
+          vars['_remember'] = ''
+          vars['submitLogin'] = URI.encode_www_form_component('Log in')
+          vars['publicKey'] = public_key
+          vars['encryptedPassword'] = Rex::Proto::Teamcity::Rsa.encrypt_data(password, public_key)
+
+          vars.each.map { |key, value| "#{key}=#{value}" }.join('&')
+        end
+
+        # Create a login request for the provided credentials.
+        # @param [String] username The username to create the login request for.
+        # @param [String] password The password to log in with.
+        # @param [String] public_key The public key to encrypt the password with.
+        # @return [Hash] The login request parameter hash.
+        def create_login_request(username, password, public_key)
+          {
+            'method' => 'POST',
+            'uri' => normalize_uri(@uri.to_s, SUBMIT_PAGE),
+            'ctype' => 'application/x-www-form-urlencoded',
+            'data' => create_login_request_body(username, password, public_key)
```

Comment:
Does vars_post work? 🤔 

```suggestion
            'vars_post' => {
              username: username,
              ... etc ...
```